### PR TITLE
CurrencySelector #70

### PR DESCRIPTION
**Pull request by K-w-e**
I managed the currency selection. 
I created a basic provider.
Then, obtaining the value through it, I display the selected currency within the application. 
Additionally, some StatelessWidget have been converted to ConsumerWidget just to read the selected currency value. 

Feel free to provide any suggestions or requests for modifications!

### CODE REVIEW

File: lib/providers/currency_provider.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import '../constants/constants.dart';
```
Comment (jackrua) [inline lib/providers/currency_provider.dart:4]:
```suggestion

```

**Review by jackrua — COMMENTED** [review]
Wonderful job! this currency thing it's something I had on my issue list for a while. Tried this PR on my phone and it seems that everything works as it should. 

Just to be pedantic, my Editor tells me that this is an unused import.

File: lib/providers/currency_provider.dart (hunk 1 above)
Comment (K-w-e) [reply lib/providers/currency_provider.dart:4]: Oh you are right, i will fix it!

**Review by theperu — APPROVED** [review]
Thanks for taking the time to do this, it was probably quite boring 😅
It seems to be working as expected and I think we can merge

**Pull request merged**